File: frontend/src/models/studentquestion/StudentQuestion.ts
import Question from '@/models/management/Question';
import Evaluation from '@/models/studentquestion/Evaluation';

export default class StudentQuestion {
  id!: number;
  question!: Question;
  evaluation!: Evaluation;
  student!: number;
  constructor(jsonObj?: StudentQuestion) {
    if (jsonObj) {
      this.id = jsonObj.id;

      if (jsonObj.evaluation)
        this.evaluation = new Evaluation(jsonObj.evaluation);
      this.question = new Question(jsonObj.question);
      this.student = jsonObj.student;
    }
  }
}

File: backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/studentquestion/domain/StudentQuestion.java
package pt.ulisboa.tecnico.socialsoftware.tutor.studentquestion.domain;

import pt.ulisboa.tecnico.socialsoftware.tutor.question.domain.Question;
import pt.ulisboa.tecnico.socialsoftware.tutor.user.User;

import javax.persistence.*;

@Entity
@Table(name = "student_questions")
public class StudentQuestion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User student;

    @OneToOne
    @JoinColumn(name = "question_id")
    private Question question;

    @OneToOne(mappedBy = "studentQuestion")
    private Evaluation evaluation;

    public StudentQuestion() {

    }

    public StudentQuestion(User user, Question question) {
        this.student = user;
        this.question = question;
    }

    public Evaluation getEvaluation() {
        return evaluation;
    }

    public void setEvaluation(Evaluation evaluation) {
        this.evaluation = evaluation;
    }

    public Question getQuestion() {
        return question;
    }

    public User getStudent() {
        return student;
    }

    public Integer getId() {
        return this.id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public void setStudent(User student) {
        this.student = student;
    }

    public void setQuestion(Question question) {
        this.question = question;
    }
}
